command/repo.go

```diff
@@ -63,6 +83,162 @@ func repoClone(cmd *cobra.Command, args []string) error {
 	return utils.PrepareCmd(cloneCmd).Run()
 }
 
+func isURL(arg string) bool {
+	return strings.HasPrefix(arg, "http:/") || strings.HasPrefix(arg, "https:/")
+}
+
+var Since = func(t time.Time) time.Duration {
+	return time.Since(t)
+}
+
+func repoFork(cmd *cobra.Command, args []string) error {
+	ctx := contextForCommand(cmd)
+
+	clonePref, err := cmd.Flags().GetString("clone")
+	if err != nil {
+		return err
+	}
+	remotePref, err := cmd.Flags().GetString("remote")
+	if err != nil {
+		return err
+	}
+
+	apiClient, err := apiClientForContext(ctx)
+	if err != nil {
+		return fmt.Errorf("unable to create client: %w", err)
+	}
+
+	var toFork ghrepo.Interface
+	inParent := false // whether or not we're forking the repo we're currently "in"
+	if len(args) == 0 {
+		baseRepo, err := determineBaseRepo(cmd, ctx)
+		if err != nil {
+			return fmt.Errorf("unable to determine base repository: %w", err)
+		}
+		inParent = true
+		toFork = baseRepo
+	} else {
+		repoArg := args[0]
+
+		if isURL(repoArg) {
+			parsedURL, err := url.Parse(repoArg)
+			if err != nil {
+				return fmt.Errorf("did not understand argument: %w", err)
+			}
+
+			toFork, err = ghrepo.FromURL(parsedURL)
+			if err != nil {
+				return fmt.Errorf("did not understand argument: %w", err)
+			}
+
+		} else {
+			toFork = ghrepo.FromFullName(repoArg)
+			if toFork.RepoName() == "" || toFork.RepoOwner() == "" {
```

Comment:
I wonder if there should exist a shared variant of `FromFullName` that raises an error like this so we can consistently parse user input with that

Reply:
I do think FromFullName should be stricter; I can't think of a time when an empty owner or name is anything but an error.